### DISCUSSION TITLE
Clarification on "input" and "id"

### DISCUSSION THREAD

**Discussion by sockmonkey0223** [emporia-vue-local/esphome#201] [Q&A]
As I build my configuration, I just want to make sure my understanding is correct regarding "input" and "id".

Am I correct in assuming that the "input" refers to the actual physical port on the Vue, and that "id" is an internal label or reference?

e.g.
My HVAC compressor is on breaker or circuit number 36. I plug the CT into port 1 on the Vue. Hence:

 - { phase_id: phase_a, input:  "1", power: { id:  cir36, filters: [ *pos, multiply: 2 ] } }
 - { platform: copy, name: "HVAC Compressor Power", source_id:  cir36, filters: *throttle_avg }
 - { power_id:  cir36, platform: total_daily_energy, accuracy_decimals: 0, name:  "HVAC Compressor Daily Energy", filters: *throttle_time }

**Comment by flaviut** [comment]
You are correct :)

**Comment by sockmonkey0223** [comment]
A couple more questions:

1) Is there any advantage to using 2 lines (per the example) vs 1 to create the sensor?
     - { phase_id: phase_a, input:  "1", power: { id: cir2, filters: [ *pos, multiply: 2 ] } }
     - { platform: copy, name: "Range Power", source_id:  cir2, filters: *throttle_avg }
     
     vs.

     - { phase_id: phase_a, input:  "1", power: { name: "Range Power". id: cir2, filters: [ *throttle_avg, *pos, multiply: 2 ] } }

2) The readme says "Note the sliding_window_moving_average" but I don't see that in the example config. I've also seen other configs using moving_average. Which of the three are currently recommended?

**Reply by flaviut** [reply]
> Is there any advantage to using 2 lines (per the example) vs 1 to create the sensor?

Yup! Context: https://github.com/emporia-vue-local/esphome/pull/200, https://www.technowizardry.net/2023/02/local-energy-monitoring-using-the-emporia-vue-2/

Would suggest [reading the blog post around](https://www.technowizardry.net/2023/02/local-energy-monitoring-using-the-emporia-vue-2/) "Previously, esphome would perform a 2.88s moving average, then integrate that value for energy consumption"--it explains it much better than I can.

In short, doing it this way leads to improved energy measurement accuracy.

> Note the sliding_window_moving_average" but I don't see that in the example config

Fixed :)

> Which of the three are currently recommended?

The main page will be always be updated to contain the best practices.
